src/main/java/io/cdap/http/internal/ForwardingEventExecutor.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.http.internal;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ProgressivePromise;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.concurrent.ScheduledFuture;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A {@link EventExecutor} that forwards all methods to another {@link EventExecutor}.
+ */
+public class ForwardingEventExecutor implements EventExecutor {
```

Comment:
Should it `implement OrderedEventExecutor`? All our executors are `OrderedEventExecutor` and this is marker interface that can be used by netty, so it's better not to lose it

Reply:
I would keep the "Forwarding" class purely for forwarding, but we can have the sub-class (inside the `NettyHttpService`) to also implement the tagging `OrderedEventExecutor` interface since it knows it is using the `NonStickyEventExecutorGroup` there.

Reply:
I mean there are checks within netty if the executor implements OrderedEventExecutor or not. By using this delegate we are dropping this marker, not sure what would it affect